vendor/github.com/containerd/containerd/platforms/cpuinfo.go

```diff
@@ -0,0 +1,122 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package platforms
+
+import (
+	"bufio"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+// Present the ARM instruction set architecture, eg: v7, v8
+var cpuVariant string
+
+func init() {
+	if isArmArch(runtime.GOARCH) {
+		cpuVariant = getCPUVariant()
```

Comment:
this makes calls that can return errors and log... is that expected on startup for the kubelet (it could happen when running things like `kubelet --help`)?

Reply:
Thanks for looking through the dep changes.

With regards to this `init()` function, looks like this was refactored in https://github.com/containerd/containerd/pull/4953 and moved out of init(), but the change isn't in containerd v1.4.4 (only in unreleased containerd v1.5).

It looks like the main place error can happen here is if it can't open `/proc/cpuinfo` (https://github.com/containerd/containerd/blob/release%2F1.4/platforms/cpuinfo.go#L49-L53) which is hopefully unlikely :). Not clear what we can do here unfortunately considering we can't move to containerd v1.5 yet...

Reply:
> It looks like the main place error can happen here is if it can't open /proc/cpuinfo (https://github.com/containerd/containerd/blob/release%2F1.4/platforms/cpuinfo.go#L49-L53) which is hopefully unlikely

if this code path happens on Windows, there is no /proc/ there.

Reply:
my comment was more of a remark that code should be written as preemptively OS-portable.

k8s doesn't produce ARM binaries for Windows. by the time (and if) we start doing that, the containerd version will likely be bumped to not have this init() code.